spec/helpers/tests2png/painter.js

```diff
@@ -0,0 +1,271 @@
+      }
+      break;
+    case 'E': out = drawError(out, x, y, startX, angle); break;
+    case 'C': out = drawComplete(out, x, y, maxFrame, angle, streamData); break;
+    default: break;
+    }
+  });
+  return out;
+}
+
+function drawObservable(out, maxFrame, y, streamData) {
+  var offsetY = OBSERVABLE_HEIGHT * 0.5;
+  var angle = 0;
+  out = drawObservableArrow(out, maxFrame, y + offsetY, angle, streamData);
+  out = drawObservableMessages(out, maxFrame, y + offsetY, angle, streamData);
+  return out;
+}
+
+function drawOperator(out, label, y) {
+  out = out.stroke('#000000', 3);
+  out = out.fill('#FFFFFF00');
+  out = out.drawRectangle(
+    CANVAS_PADDING, y,
+    CANVAS_WIDTH - CANVAS_PADDING, y + OPERATOR_HEIGHT);
+  out = out.strokeWidth(-1);
+  out = out.fill('#000000');
+  out = out.font('helvetica', 54);
+  out = out.draw(
+    'translate 0,' + (y + OPERATOR_HEIGHT * 0.5 - canvasHeight * 0.5),
+    'gravity Center',
+    'text 0,0',
+    String('"' + label + '"'));
+  return out;
+}
+
+function sanitizeHigherOrderInputStreams(inputStreams, outputStream) {
+  // Remove cold inputStreams which are already nested in some higher order stream
+  return inputStreams.filter(function (inputStream) {
+    return !inputStreams.concat([outputStream]).some(function (otherStream) {
+      return otherStream.messages.some(function (msg) {
+        var passes = isNestedStreamData(msg) &&
+          inputStream.cold &&
+          _.isEqual(msg.notification.value.messages, inputStream.cold.messages);
+        if (passes) {
+          if (inputStream.cold.subscriptions.length) {
+            msg.notification.value.subscription = {
+              start: inputStream.cold.subscriptions[0].subscribedFrame,
+              end: inputStream.cold.subscriptions[0].unsubscribedFrame
+            };
+          }
+        }
+        return passes;
+      });
+    });
+  });
+}
+
+
+module.exports = function painter(inputStreams, operatorLabel, outputStream, filename) {
+  var maxFrame = getMaxFrame(inputStreams.concat(outputStream));
+  inputStreams = sanitizeHigherOrderInputStreams(inputStreams, outputStream);
+  var inputStreamsHeight = inputStreams
+    .map(measureStreamHeight(maxFrame))
+    .reduce(function (x, y) { return x + y; }, 0);
+  canvasHeight = inputStreamsHeight + OPERATOR_HEIGHT + measureStreamHeight(maxFrame)(outputStream);
+
+  var heightSoFar = 0;
+  var out;
+  out = gm(CANVAS_WIDTH, canvasHeight, '#ffffff');
+  inputStreams.forEach(function (streamData) {
+    out = drawObservable(out, maxFrame, heightSoFar, streamData);
+    heightSoFar += measureStreamHeight(maxFrame)(streamData);
+  });
+  out = drawOperator(out, operatorLabel, heightSoFar);
+  heightSoFar += OPERATOR_HEIGHT;
+  out = drawObservable(out, maxFrame, heightSoFar, outputStream);
+
+  out.write('./img/' + filename + '.png', function (err) {
+    if (err) {
+      return console.error(arguments);
+    }
+    //console.log(this.outname + ' created  :: ' + arguments[3]);
```

Comment:
Yes, console.log part. My 'personal' preference is do not leave unused code as commented in general. Since it's generally harmless output, what about simply enable it as verbose output?